domain/lifewatcher.go

```diff
@@ -0,0 +1,102 @@
+// Copyright 2023 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package domain
+
+import (
+	"context"
+
+	"github.com/juju/collections/set"
+	"github.com/juju/errors"
+
+	"github.com/juju/juju/core/changestream"
+	coredatabase "github.com/juju/juju/core/database"
+	"github.com/juju/juju/core/logger"
+	"github.com/juju/juju/core/watcher/eventsource"
+	"github.com/juju/juju/domain/life"
+)
+
+// LifeGetter is a function which looks up life values of the entities with the specified IDs.
+type LifeGetter func(ctx context.Context, db coredatabase.TxnRunner, ids ...string) (map[string]life.Life, error)
```

Comment:
```suggestion
type LifeGetter func(ctx context.Context, db coredatabase.TxnRunner, ids []string) (map[string]life.Life, error)
```

You need at least 1 id for this call to make sense. Let's offer strong guarantees with the type.

Reply:
ok